lib/cake_pay/src/models/cake_pay_card.dart

```diff
@@ -27,27 +26,41 @@ class CakePayCard {
     this.cardImageUrl,
     this.country,
     required this.fiatCurrency,
-    required this.denominationsUsd,
-    required this.denominations,
     this.minValueUsd,
     this.maxValueUsd,
     this.minValue,
     this.maxValue,
-  });
+    List<Denomination>? denominationItems,
+  }) : denominationItems = denominationItems ?? const [];
 
   factory CakePayCard.fromJson(Map<String, dynamic> json) {
-
     final name = stripHtmlIfNeeded(json['name'] as String? ?? '');
     final description = stripHtmlIfNeeded(json['description'] as String? ?? '');
     final termsAndConditions = stripHtmlIfNeeded(json['terms_and_conditions'] as String? ?? '');
     final howToUse = stripHtmlIfNeeded(json['how_to_use'] as String? ?? '');
-
     final fiatCurrency = FiatCurrency.deserialize(raw: json['currency_code'] as String? ?? '');
 
-    final List<String> denominationsUsd =
-        (json['denominations_usd'] as List?)?.map((e) => e.toString()).toList() ?? [];
-    final List<String> denominations =
-        (json['denominations'] as List?)?.map((e) => e.toString()).toList() ?? [];
+    List<Denomination> parsedDenomination = [];
+    final rawDenomination = json['denominations'];
+
+    if (rawDenomination is List) {
+      if (rawDenomination.isNotEmpty && rawDenomination.first is Map) {
+        parsedDenomination = rawDenomination
+            .whereType<Map<dynamic, dynamic>>()
+            .map<Denomination>((e) => Denomination.fromJson(Map<String, dynamic>.from(e)))
+            .toList();
+      } else {
+        parsedDenomination = rawDenomination
+            .map((e) => Denomination(value: Denomination._toDouble(e) ?? 0))
+            .toList();
+      }
+    } else if (rawDenomination is Map) {
+      final desc = rawDenomination['description'];
+      if (desc is List) {
+        parsedDenomination =
+            desc.map((e) => Denomination(value: Denomination._toDouble(e) ?? 0)).toList();
+      }
```

Comment:
with the new backend the denominations is always a List of objects, why are we adding all these checks? let's simplify it please, since we made the new backend to simplify things not to complicate it

Reply:
removed unnecessary checks